### PR TITLE
unigine-tropics: init at 1.3

### PR DESCRIPTION
**Pull request by barinov274**
###### Description of changes

I want to add a new package unigine-tropics. It's a benchmarking tool from unigine. Nixos users will be able to install this package on their computers and benchmark gpu.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux

### CODE REVIEW

File: pkgs/applications/graphics/unigine-tropics/default.nix [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  desktopItem = makeDesktopItem {
+    name = "Tropics";
+    exec = "Tropics";
+    genericName = "A GPU Stress test tool from the UNIGINE";
+    icon = "Tropics";
+    desktopName = "Tropics Benchmark";
+  };
+in
+stdenv.mkDerivation
+{
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-tropics/default.nix:28]:
```suggestion
stdenv.mkDerivation {
```

File: pkgs/applications/graphics/unigine-tropics/default.nix [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  libPath = lib.makeLibraryPath [
+    libglvnd
+    openal
+    glibc
+  ];
+
+  installPhase =
+    ''
+      sh $src --target $name
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-tropics/default.nix:45]:
```suggestion
  installPhase = ''
    bash $src --target $name
```

File: pkgs/applications/graphics/unigine-tropics/default.nix [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      mkdir -p $out/share/icons/hicolor
+      mkdir -p $out/share/applications/
+
+      install -m 0755 $name/bin/libUnigine_x86.so $out/lib/unigine/tropics/bin
+      install -m 0755 $name/bin/Tropics $out/lib/unigine/tropics/bin
+      install -m 0755 $name/1024x768_windowed.sh $out/bin/Tropics
+
+      cp -R $name/data $out/lib/unigine/tropics
+
+      wrapProgram $out/bin/Tropics --prefix LD_LIBRARY_PATH : $libPath:$out/lib/unigine/tropics/bin --run "cd $out/lib/unigine/tropics"
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-tropics/default.nix:58]:
```suggestion
      wrapProgram $out/bin/Tropics \
        --prefix LD_LIBRARY_PATH : $libPath:$out/lib/unigine/tropics/bin \
        --chdir $out/lib/unigine/tropics
```

File: pkgs/applications/graphics/unigine-tropics/default.nix [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      convert $name/icon.png -fill white -stroke white -draw "polygon  69.2564,84.1261 117.9,84.1261 117.9,206.56 138.1,206.56 138.1,84.1261 186.744,84.1261 186.744,65.9877 69.2564,65.9877 69.2564,84.1261" $name/icon.png
+
+      convert $name/icon.png -resize 128x128 $out/share/icons/Tropics.png
+      for RES in 16 24 32 48 64 128 256
+      do
+          mkdir -p $out/share/icons/hicolor/"$RES"x"$RES"/apps
+          convert $name/icon.png -resize "$RES"x"$RES" $out/share/icons/hicolor/"$RES"x"$RES"/apps/Tropics.png
+      done
+
+      ln -s ${desktopItem}/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-tropics/default.nix:70]: Please use copyDesktopItems.

File: pkgs/applications/graphics/unigine-tropics/default.nix [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  nativeBuildInputs =
+    [
+      autoPatchelfHook
+      makeWrapper
+      imagemagick
+    ];
+
+  buildInputs =
+    [
+      stdenv.cc.cc
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-tropics/default.nix:82]:
```suggestion
  nativeBuildInputs = [
    autoPatchelfHook
    makeWrapper
    imagemagick
  ];

  buildInputs = [
    stdenv.cc.cc
```

File: pkgs/applications/graphics/unigine-tropics/default.nix [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      libXrandr
+      libXinerama
+    ];
+
+  dontUnpack = true;
+
+  meta =
+    {
+      description = "The Unigine Heaven GPU benchmarking tool";
+      homepage = "https://benchmark.unigine.com/tropics";
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/unigine-tropics/default.nix:94]:
```suggestion
  meta = {
    description = "The Unigine Heaven GPU benchmarking tool";
    homepage = "https://benchmark.unigine.com/tropics";
```

**Review by SuperSandro2000 — COMMENTED** [review]
How big is the similarity to normal unigine? Could we do a common.nix and derive both packages from it?

File: pkgs/applications/graphics/unigine-tropics/default.nix (hunk 3 above)
Comment (barinov274) [reply pkgs/applications/graphics/unigine-tropics/default.nix:58]: Builder called die: makeWrapper doesn't understand the arg --chdir
There is no argument chdir for makeWrapper or wrapProgram 
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/make-wrapper.sh

Comment (SuperSandro2000) [reply pkgs/applications/graphics/unigine-tropics/default.nix]: There is but it is still stuck in staging https://github.com/NixOS/nixpkgs/commit/10479e4f51ff182651ccff4a71b2a94ab93816c0

**Pull request merged**